File: src/main/java/am/adrian/dungeonkeeper/game/GameStateService.java
package am.adrian.dungeonkeeper.game;

import am.adrian.dungeonkeeper.common.constant.GameState;
import org.springframework.stereotype.Service;

@Service
public class GameStateService {

    private volatile GameState state = GameState.NOT_STARTED;

    public synchronized void setState(GameState state) {
        this.state = state;
    }

    public synchronized boolean isRunning() {
        return !(state == null || state == GameState.NOT_STARTED || state == GameState.FINISHED);
    }
}

File: src/main/kotlin/am/adrian/dungeonkeeper/game/GameStateService.kt
package am.adrian.dungeonkeeper.game

import am.adrian.dungeonkeeper.constant.GameState
import org.springframework.stereotype.Service

@Service
class GameStateService {

    private var state: GameState = GameState.NOT_STARTED

    fun setState(state: GameState) {
        this.state = state
    }

    fun isStarted(): Boolean {
        return state == GameState.STARTED
    }
}
